openqabot/types/incidents.py

```diff
@@ -96,35 +96,37 @@ def __call__(
                             inc.id,
                         )
                         continue
+
+                    if inc.staging:
+                        continue
+
+                    flavor_settings = self.settings.copy()
+                    if "flavor_settings" in data:
+                        flavor_settings.update(data["flavor_settings"])
+
+                    if "packages" in data:
```

Comment:
how this related to introduction of `flavor_settings` ?

Reply:
This code has just be moved not added. 

As `flavor_settings` is updating `settings` I reorganized the code to have all stuff with different purpose as separated and sequentially ordered as possible. I can recognize at least 3 main category.

* check something and skip to schedule a POST, me moving https://github.com/openSUSE/qem-bot/pull/145/files#diff-81a88cee51bd2524a7e938a6b6d53d79f526ffb436071ead2d3d83163b38b1ecR100 is in this category  Any stuff belonging to this category can goes first
* settings related or dependent stuff
* everything else

Generally speaking this method seems to me quite long that does not help about understanding what is going one.

Reply:
any chance to do it in separate PR ?  we introducing quite big functional change here . IMHO it would be nice not to mix this with ANY other changes , so in case something will go wrong we can more easily detect the problem . If you will fully agree that this code may and should be improved . I just don't like to mix this **refactoring** together with introduction of `flavor_settings`